scripts/40-upload_to_strava.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python3
+#
+# Code by Matheus Cansian <dev@drpexe.com>
+#
+# This helper uses stravalib to send the fit files to Strava
+#
+# To install stravalib
+#
+# sudo pip install stravalib
+#
+# This script uses a third party API to facilitate Strava's OAuth
+# authentication. If you want to deploy your own API you can use
+# the instructions below and change the constants on this script.
+# https://github.com/mscansian/drpexe-uploader
+#
+# You can fetch the credentials by running the script without any arguments
+#
+# ./40-upload_to_strava.py
+#
+# Credentials are written to ~/.drpexe-uploader-credentials
+#
+# Don't forget to make this script executable :
+#
+# chmod +x /path/to/40-upload_to_strava.py
+from __future__ import print_function, with_statement
+
+import sys
+import os.path
+
+try:
+    from urllib.parse import urlparse
+    from http.server import HTTPServer, BaseHTTPRequestHandler
+except ImportError:  # Python 2.7
+    from urlparse import urlparse
+    from BaseHTTPServer import HTTPServer, BaseHTTPRequestHandler
+
+from stravalib import Client
+from stravalib.exc import ActivityUploadFailed
+
+# drpexe-uploader config
+# https://github.com/mscansian/drpexe-uploader
+# do not change unless you want to create your own strava app
+DRPEXE_CLIENT_ID = 17666
+DRPEXE_UPLOADER_API = (
+    'https://in27m0omnk.execute-api.us-east-1.amazonaws.com/prod'
+)
+
+STRAVA_CREDENTIALS_FILE = os.path.expanduser('~/.drpexe-uploader-credentials')
+STRAVA_UPLOAD_PRIVATE = False
+LOCAL_SERVER_PORT = 8000
+
+
+def main(action, filename):
+    if action != "DOWNLOAD":
+        return 0
+
+    try:
+        with open(STRAVA_CREDENTIALS_FILE, 'r') as f:
+            access_token = f.read().strip(' \t\n\r')
+    except FileNotFoundError:
+        print('No Strava credentials provided.')
+        print('You first need to run the script to fetch the credentials')
+        print('./40-upload_to_strava.py')
+        return -1
+
+    try:
+        client = Client(access_token=access_token)
+        print('Uploading {}: '.format(os.path.basename(filename)), end='')
+        with open(filename, 'rb') as f:
+            upload = client.upload_activity(
+                activity_file=f,
+                data_type='fit',
+                private=STRAVA_UPLOAD_PRIVATE,
+            )
+    except (ActivityUploadFailed, FileNotFoundError) as err:
+        print('FAILED')
+        print('Reason:', err)
+        return -1
+
+    print('SUCCESS')
+    return 0
+
+
+def start_strava_auth_flow():
+    print('---------------------------------------------')
```

Comment:
I'm a sucker for the https://en.wikipedia.org/wiki/Box_Drawing unicode block, but yeah :) :+1: